### PR TITLE
Creates TIP explaining usage of HtmlAttributeName in custom Tag Helper

### LINKED ISSUE #15626: Explain how to use the HtmlAttributeName attribute
Unless I've missed it, I don't think it's covered. I'd like a simple example, something like:

```csharp
[HtmlTargetElement("div")]
public class MyTagHelper : TagHelper
{
    [HtmlAttributeName("asp-do-thing")]
    public bool? DoThing { get; set; }

    public override Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
    {
        // some logic querying value of DoThing
    }
}
```

I appreciate the doc already covers passing a model object instance via a TagHelper, but from what I understand Razor is just looking to see that the property and attributes names match up. So it would be useful to show how to override this.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 19ecfbee-8b84-9ef2-65b6-8c2542f6370f
* Version Independent ID: 9a0cbaa2-65c9-1618-1003-5fd2e7f6192f
* Content: [Author Tag Helpers in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/views/tag-helpers/authoring?view=aspnetcore-3.0)
* Content Source: [aspnetcore/mvc/views/tag-helpers/authoring.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/mvc/views/tag-helpers/authoring.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Ducki**
This PR creates a TIP box in the "Create Tag Helpers" chapter ("SetAttribute and SetContent" section) hinting at the possibility to map arbitrary attribute names to a differently named property via the `HtmlAttributeName` attribute.

Also fixes #15626.

### CODE REVIEW

**Comment by guardrex** [comment]
Hello @Ducki ... Thanks for sending this in. I made a quick style pass and pinged Rick. He's not a huge *TIP fan* 😆, but we'll see. We often just remove the formatting and just have this type of content as normal paragraph/example content.

File: aspnetcore/mvc/views/tag-helpers/authoring.md [hunk 1]
```diff
@@ -94,6 +94,22 @@ Update the `EmailTagHelper` class with the following:
...
+  > [HtmlAttributeName("recipient")]
+  > public string? MailTo { get; set; }
+  > ```
+  >
+  > Tag Helper for the `recipient` attribute:
+  >
+  > ```html
+  > <email recipient="…"/>
+  > ```
+
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/views/tag-helpers/authoring.md:112]:
```suggestion
```

**Comment by Rick-Anderson** [comment]
Sorry, I moved your comment below and it killed the PR. You need to add the following
```suggestion
  A different attribute name can be mapped to a property using the [`[HtmlAttributeName]` attribute]  (xref:Microsoft.AspNetCore.Razor.TagHelpers.HtmlAttributeNameAttribute). To map an attribute named   `recipient` to the `MailTo` property:
  
  ```csharp
  [HtmlAttributeName("recipient")]
  public string? MailTo { get; set; }
  ```
  
  Tag Helper for the `recipient` attribute:
  
  ```html
  <email recipient="…"/>
  ```

```
Add that just above `## ProcessAsync`

**Comment by Ducki** [comment]
Got it, done :)

**Review by Rick-Anderson — APPROVED** [review]

**Comment by Rick-Anderson** [comment]
Thanks for the nice PR @Ducki , keep them coming.
Be sure to check out your name as a contributor about the 3rd of next month on the [What's new page](https://docs.microsoft.com/en-us/aspnet/core/whats-new/) in the **Community contributors** section at the bottom.

**Pull request merged**